### PR TITLE
Use assertEquals for comparing reference lists in test

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: tests/unit/ReferenceListTest.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function getConstructorArg() {
 	public function testCanConstructWithReferenceListObject() {
 		$original = new ReferenceList( array( new Reference( array( new PropertyNoValueSnak( 1 ) ) ) ) );
 		$copy = new ReferenceList( $original );
-		$this->assertSame( 1, $copy->count() );
+		$this->assertEquals( $original, $copy );
```
Comment (thiemowmde) [inline tests/unit/ReferenceListTest.php]: This uses PHP's dark magic of comparing objects. I wanted to avoid this when I wrote the test.

Comment (JeroenDeDauw) [inline tests/unit/ReferenceListTest.php]: I was under the impression that PHPUnit compares objects in a sane way. We are certainly using assertEquals for this in other places.

Comment (thiemowmde) [inline tests/unit/ReferenceListTest.php]: Define "sane". PHPUnit does not add much magic on top of PHP's. It delegates the object comparison to PHP. We already run into horrifying problems when comparing list objects that extend one of PHP's classes. I'm very alarmed whenever I see something like this. I will not merge this PR.

**Comment by thiemowmde** [comment]
I reworked the test in #441, please have a look.

**Pull request closed without merging**